client/packages/plugins/backendCommon/utils.ts

```diff
@@ -0,0 +1,55 @@
+const wrapSql = (fields: string[], sqlStatement: string) => {
+  if (sql_type() === 'sqlite') {
+    return `
+            SELECT json_object(${fields.map(field => `'${field}', ${field}`)}) AS json_row 
+            FROM (${sqlStatement})
+            `;
+  }
+
+  if (sql_type() === 'postgres') {
+    return `
+            SELECT json_agg(inner) AS json_row 
+            FROM (${sqlStatement}) AS inner
+            `;
+  }
+
+  throw new Error('Unknown sql_type');
+};
+
+export const sqlQuery = <K extends string>(
+  fields: K[],
+  sqlStatement: string
+): Record<K, any>[] => {
+  const adjustedStatement = wrapSql(fields, sqlStatement);
+  return sql(adjustedStatement) as Record<K, any>[];
+};
+
+export const startOfDay = (date: Date) => {
+  const start = new Date(date);
+  start.setHours(0);
+  start.setMinutes(0);
+  start.setSeconds(0);
+  return start;
+};
+export const endOfDay = (date: Date) => {
+  const end = new Date(date);
+  end.setHours(23);
+  end.setMinutes(59);
+  end.setSeconds(59);
+  return end;
+};
+
+export const sqlDateTime = (date: Date) =>
+  // toJSON will make it utc
+  date.toJSON().replace('T', ' ').split('.')[0];
+export const localDate = (date: Date) => {
+  return `${date.getFullYear()}-${date.getMonth() + 1}-${date.getDate()}`;
+};
+export const sqlList = (list: string[]) => '("' + list.join('","') + '")';
+export const fromSqlDateTime = (datetime: string) => {
+  log(
+    JSON.stringify({ datetime, converted: `${datetime.split(' ').join('T')}Z` })
+  );
```

Comment:
was the log your testing?

Reply:
Thanks, yep: 7dc2e07449e6a1f08a8a3bb57f20aa8f8e7c2110